sky/clouds/kubernetes.py

```diff
@@ -34,6 +35,26 @@
 # E.g., FUSE device manager daemonset is run in this namespace.
 _SKYPILOT_SYSTEM_NAMESPACE = 'skypilot-system'
 
+_REQUIRED_BINARIES = ['netcat', 'socat']
+
+
+def _check_k8s_dependencies() -> Tuple[bool, Optional[str]]:
```

Comment:
We should remove or reuse kubernetes_utils.check_port_forward_mode_dependencies():
https://github.com/skypilot-org/skypilot/blob/73e95ab79bf75923963a93733e54c100d0567e02/sky/provision/kubernetes/utils.py#L1785

Also note that nc checks need some careful handling since the default mac nc doesn't work: 
https://github.com/skypilot-org/skypilot/blob/73e95ab79bf75923963a93733e54c100d0567e02/sky/provision/kubernetes/utils.py#L1827-L1843

Reply:
Probably need you guys to take over this PR